### DISCUSSION TITLE
Feature Request: Application Blacklist

### DISCUSSION THREAD

**Discussion by xNS5** [rxhanson/Rectangle#971] [Ideas]
I'd like to request an application blacklist feature be added to Rectangle. Presently, Rectangle's hotkeys work on most application windows (with a few exceptions like for context menu pop-ups. In my specific use case, there are two that I'd like rectangle to not work on: [Unclutter](https://unclutterapp.com/) and [Google Drive desktop](https://support.google.com/a/users/answer/9965580?hl=en). 

Unclutter is activated by a swipe/scroll down action from the menu bar. However, whenever it's in focus I can use my hotkeys to move the window around. Google Drive has its own toolbar menu which I'm also to move with Rectangle. 

Here is an example of Unclutter being moved by Rectangle:
<img width="1629" alt="Screen Shot 2022-10-28 at 11 10 26 PM" src="https://user-images.githubusercontent.com/27764452/198816820-d79e8d32-d81f-4710-87b3-bc45f4c10b2d.png">

I believe that this is intended behavior of Rectangle and isn't a bug.

What I propose be added is a small screen to add/remove applications from a blacklist, and a check be added when using hotkeys to see if the currently focused application is blacklisted. I don't know if this feature would interfere with window snapping, but if it does then an application-wide blacklist would work perfectly as well. 

I'm not super familiar with Swift, but I found this [Stackexchange post](https://apple.stackexchange.com/questions/123730/is-there-a-way-to-detect-what-program-is-stealing-focus-on-my-mac) which might help determine which application is currently in focus. While it's in python, `NSWorkspace.sharedWorkspace().activeApplication()['NSApplicationName']` looks promising if there's an equivalent swift API.

**Comment by rxhanson** [comment]
To disable keyboard shortcuts for specific applications, you'll need to get the menu bar icon back up and use the menu bar menu to disable the apps you don't want to use.

**Reply by xNS5** [reply]
Hah, well don't I feel sheepish. I was looking for something in the settings, I didn't realize that there was something in the menu bar icon. I think there would be a benefit to modifying the blacklisted applications from the settings, but at least there *is* a means to blacklist applications as needed!

Thanks for the reply!
